store.go

```diff
@@ -53,9 +53,7 @@ func (s *Store) Insert(record Record) error {
 		return ErrNonNewDocument
 	}
 
-	if record.GetID().IsEmpty() {
-		record.SetID(NewID())
-	}
+	record.Identify()
```

Comment:
I would keep it as it was. It's not going to be used anywhere else and feels a little pointless to keep this method just for this ad-hoc logic